### ISSUE TITLE
Add `Kotlin value class member name` to `Naming Rule Set`

### ISSUE THREAD

**Issue by igorwojda** [detekt/detekt#5883]
The rule or changes you're looking for might have already been suggested!

## Expected Behavior of the rule

Compilant
```kotlin
@JvmInline
value class ProjectId(val value: String)
```

Non-Compilant
```kotlin
@JvmInline
value class ProjectId(val id: String)
```

## Context

Since Kotlin value class is a wrapper it may be helpful to always use `value` as the name for the property holding the wrapped value.

This would avoid heaving nasty "value extraction code" eg. consider this mapper that maps the `Project` domain model to Spring `ProjectJapEntity`: 
![image](https://user-images.githubusercontent.com/530122/223116143-6224479c-d918-43b4-88b4-cb5806e6a241.png)

**Comment by cortinico** [comment]
> Since Kotlin value class is a wrapper it may be helpful to always use `value` as the name for the property holding the wrapped value.

Could you elaborate more on why this should be enforced?

**Comment by lexa-diky** [comment]
Looking into [documentation](https://kotlinlang.org/docs/inline-classes.html#inline-classes-and-delegation) and [KEEP](https://github.com/Kotlin/KEEP/blob/master/proposals/inline-classes.md) JetBrains does not have any rules / expectancies regarding value class field naming. Don't think this rule should be enforced in Detekt.

**Comment by atulgpt** [comment]
I like this idea as this creates consistency across all `value class`. We can make this optional rule as @igorwojda pointed out and later on if need arises we can make the name `value` configurable

**Comment by igorwojda** [comment]
Going a bit further with this:

- In Android, there is [LiveData](https://developer.android.com/reference/android/arch/lifecycle/LiveData#getValue()) class. This class serves as a wrapper for the value, so the value can be retrieved by using `getValue()` method, translated in Kotlin to `value`
- In reactive programming (Kotlin FLow / RxJava) stream is also a kind of wrapper for the emitted value

My point is if the class serves a is wrapper that usually provides meaningful name for wrapped value then the wrapped value can be simply named `value`

**Comment by BraisGabin** [comment]
This rule goes to the bucket of controversial rules. This is a rule that I would adopt in my projects for sure but I'm not sure if it should be inside the default rules of detekt.

I think that we would need some talks, blogs, toots/tweets or documentation talking about this pattern to make this rule inside the default rule set. Otherwise I think that a custom rule would be a perfect fit. And add it in our [marketplace](https://detekt.dev/marketplace)
